### DISCUSSION TITLE
Refactoring of CMI file & folder structure

### DISCUSSION THREAD

**Discussion by Tarlak333** [Zrips/CMI#5040] [Enhancement]
Hello,

I wanted to to see if you would be open to this request.  I notice a lot of people bash CMI as "bloat ware" (which honestly I did to at first) however once I fully engaged with what features could be turned on/off and the level of customization that CMI brought to my servers I can't live with out it.  (It's truly  a momentous piece of work!)  I've been using CMI for almost a year now and have one suggestion that I think will universally benefit every end user.... Refactoring the entire file and directory structure...I know this would be a tremendous task, that doesn't really add and "fixes" or new content.  But it would ease the current burden on the current userbase and make CMI easier to adopt for new users.  ...and lastly by streamlining the content packaging CMI would appear less "bloated" but clean, efficient, and easily manageable.

My proposed layout is essentially that way Bento-Box has their files/folders setup.

Example Images:
https://imgur.com/a/NRL4pv8

...plugins\cmi\modules\ (would contain <module>.yml, module file directory containing all the config & data files for that module)

General suggestions:
- Some things like chat, MOTD, announcement/broadcast msgs could possibly be consolidated .
- Keep others as stand alone features like Kits, Portals, Jails, Enchanting, Repair, Holograms, Heads, etc...

The presence of a module .jar in the ...\cmi\modules folder would indicate to the main CMI config that the module was to be used or not.  (This would completely eliminate the need for the modules.yml) (Or if you don't like that since some ppl might decide to use a feature later and want all the files present, just put as the first line in each individual modules config.yml "Enable=True/False".)

I'm not suggesting you adopt the fancy BentoBox custom addon download feature as shown here (https://download.bentobox.world/custom) ...although it is pretty cool...up to you.  Just the consolidated & modular file/folder structure used.

Thank you for your consideration of this suggestion,

Tarlak333

**Comment by Zrips** [comment]
This was on my mind too. Major thing is not to make it more organized but to actually create conversion tool for this action where one things would be simply moved while others needs to be taken away from inside main config file and put into separate one. Something to go wrong during this process is quite likely. Will think of some possible way to migrate without causing issues, but currently my main attention is towards CMILib which can clean up some of the things from plugin itself, after which it will be more clear what needs to be moved into modules and what goes into CMILib itself.
